database/migrations/001-punishments-schema.sql

```diff
@@ -0,0 +1,46 @@
+--------------------------------------------------------------------------------
+-- Up
+--------------------------------------------------------------------------------
+
+CREATE TABLE IF NOT EXISTS punishments (
+	punishType TEXT NOT NULL,
+	userid TEXT NOT NULL,
+	ips TEXT NOT NULL,
+	userids TEXT NOT NULL,
+	expireTime INTEGER NOT NULL,
+	reason TEXT NOT NULL,
+	PRIMARY KEY (punishType, userid)
+);
+
+CREATE TABLE IF NOT EXISTS room_punishments (
+	punishType TEXT NOT NULL,
+	-- roomid:userid
```

Comment:
Are you implying `id` is a concatenation of `roomid` and `userid`? Why not just have them as separate columns? If those columns are then indexed, this lets us look up by `roomid` and `userid` very quickly